src/tachimawari/joint/tf2/tf2_manager.cpp

```
@@ -0,0 +1,80 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "tachimawari/joint/tf2/tf2_manager.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <memory>
+#include <nlohmann/json.hpp>
+#include <string>
+#include <vector>
+
+namespace tachimawari::joint
+{
+
+Tf2Manager::Tf2Manager() {}
+
+bool Tf2Manager::load_configuration(std::string path)
+{
+  std::string ss = path + "frame_measurements.json";
+
+  std::ifstream input(ss, std::ifstream::in);
+  if (!input.is_open()) {
+    throw std::runtime_error("Unable to open `" + ss + "`!");
+  }
+
+  nlohmann::json config = nlohmann::json::parse(input);
+
+  for (auto & item : config.items()) {
+    // Get all config
+    try {
+      std::string name = item.key();
+      double translation_x = item.value().at("translation").at("x");
+      double translation_y = item.value().at("translation").at("y");
+      double translation_z = item.value().at("translation").at("z");
+      double const_roll = item.value().at("const_rpy").at("roll");
+      double const_pitch = item.value().at("const_rpy").at("pitch");
+      double const_yaw = item.value().at("const_rpy").at("yaw");
+      auto map_string_id = FrameId::frame_string_id.find(name);
+
+      uint8_t id = map_string_id->second;
+      frames.push_back(
+        Frame(id, translation_x, translation_y, translation_z, const_roll, const_pitch, const_yaw));
+    } catch (nlohmann::json::parse_error & ex) {
+      std::cerr << "parse error at byte " << ex.byte << std::endl;
+    }
+  }
```

Better to throw the exception instead of letting the program running with unknown behavior.
```suggestion
    } catch (nlohmann::json::parse_error & ex) {
      std::cerr << "parse error at byte " << ex.byte << std::endl;
      throw ex;
    }
  }
```